portable/GCC/ARM_CM0/portasm.c

```diff
@@ -456,11 +456,11 @@ void vClearInterruptMask( __attribute__( ( unused ) ) uint32_t ulMask ) /* __att
             "                                \n"
             " stack_on_psp:                  \n"
             "     mrs r0, psp                \n"
-            "     b route_svc                \n"
+            "     bl route_svc               \n"
```

Comment:
From [Armv6-M architecture manual](https://developer.arm.com/documentation/ddi0419/latest/), section A4.3 Branch instructions, the range supported by `B` instruction is +/- 2KB which is not sufficient to jump to `vPortSVCHandler_C` in the [issue](https://github.com/FreeRTOS/FreeRTOS-Kernel/issues/1130) you have raised. Changing from instruction `B` to `BL` solves this issue. Please note that `B` is a 16-bit thumb instruction, where as `BL` is 32-bit thumb instruction. The `route_svc` label is is in the same function and is just after couple of instructions and it should still be reachable with `B` instruction and save 2 bytes of code, unless I'm missing something. Same suggestion for the other change involving `route_svc`.

Reply:
Hi, Okay for the `route_svc` jump the `B` should be sufficient in this case. I will update.
Thanks!